docs/hugo/content/contributing/generator-overview.md

```diff
@@ -0,0 +1,49 @@
+# Code Generator Overview
+
+Core to Azure Service Operator (ASO) v2 is our code generator. This consumes ARM JSON Schema and Swagger specifications and generates the code for each desired resource.
+
+## Code Structure
+
+The key packages used to structure the code of the generator are as follows:
+
+| Package          | Content                                                                                                                                               |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `astmodel`         | Short for _Abstract Syntax Tree Model_, this package contains the core data types for defining the Go functions, data types, and methods we generate. |
+| `functions`        | Support for generation of individual functions, based on the `astmodel.Functions` interface                                                           |
+| `interfaces`       | Support for generation of interface implementations, based on the `astmodel.InterfaceImplementer` interface                                           |
+| `testcases`        | Support for generation of test cases (used to verify our generated code works as expected), based on the `astmodel.TestCase` interface                |
+| `astbuilder`       | Intention revealing utility methods for creating the underlying Go abstract syntax tree we serialize as the last step of generation.                  |
+| `codegen`          | The core processing pipeline of the code generator                                                                                                    |
+| `codegen/pipeline` | Individual pipeline stages that are composed to form the code generator itself.                                                                       |
+| `test`             | Support methods to make writing tests easier                                                                                                          |
+
+### Object Model
+
+At the core, the code generator works with a rich semantic object model that captures the structure of the resources (and related types) we are generating.
+
+Underpinning this object model is the `astmodel` package. One of the core `astmodel` interfaces is `Type`, representing a specific data type. Implementations of `Type` fall into a few separate groups:
+
+* Simple Go types, including `PrimitiveType`, `ArrayType`, and `MapType`.
+* Complex types with internal structure, most notably `ObjectType` and `ResourceType`.
+* Wrapper types that provide additional semantics, including `OptionalType`, `ErroredType`, `FlaggedType` and `ValidatedType`.
+* Schema types, including `AllOfType` and `OneOfType`.
```

Comment:
Adding more information on `AllOfType` and `OneOfType` would be helpful to get better understanding of these types.

Reply:
Done.